src/System.Text.Json/tests/JsonReaderTests.cs

```diff
@@ -1,5 +1,6 @@
 ﻿using System.Collections.Generic;
```

Comment:
There was limited focus on invalid JSON because I was focused on correctness and under the assumption that the JSON is valid. I think we need to specify how to handle some of the error cases that you mentioned and thoroughly tests those scenarios. Regarding the case when the buffer is too short, do we allocate a larger buffer and copy over the data?